### PR TITLE
Optimize Arch Linux package

### PR DESCRIPTION
**Pull request by quyse**
This is attempt to make itch Arch Linux package depend on official `electron` package, rather than include full-blown electron binaries inside it. I'm not sure how bad this idea is (i.e. what if version of electron in Arch repo is not what itch is expecting?), but it cuts the size of resulting package significantly, and the result seems to be working perfectly on my machine.

Before:
itch-23.2.1-1-x86_64.pkg.tar.xz - size 43 Mb
unpacked size shown by pacman: Total Installed Size:  161.88 MiB

After:
itch-23.2.1-1-x86_64.pkg.tar.xz - size 13 Mb
unpacked size shown by pacman: Total Installed Size:  48.27 MiB

Basically I remove dev dependencies from node_modules (including electron), and pack the app manually instead of using `electron-packager`.

Script is a bit hacky, I can clean it up if you are OK with general idea :)

### CODE REVIEW

**Comment by fasterthanlime** [comment]
Well, that's an interesting approach! I'm not sure how it'll fit into my itchSetup plans (cf. https://github.com/itchio/itch/issues/1093) though.

I can see at least one easy improvement to the script: use `"$@"` rather than `$*`, see https://google.github.io/styleguide/shell.xml

**Comment by fasterthanlime** [comment]
So, now I have a clearer picture and unfortunately I don't think I can accept this PR.

Yes, using the distribution's electron package *is* going to save you about 40 megs, if electron is already installed for other apps.

However, electron is a fast-moving target: there's multiple releases per month, and I follow them closely. It's not unusual for me to switch to a more recent version just to get access to a feature that makes the UI/UX better for itch users, so:

  * if the distro packages lags behind by one day, that's one day of the app being broken
  * conversely, if the distro packages are extremely up-to-date, they might be for a version that deprecates some of the APIs the itch app is still using.
  * (correct me if I'm wrong, but I'm fairly certain arch doesn't allow multiple versions of a package to be installed at the same time to satisfy different apps - what usually happens, much like in debian, is splitting a library into different packages for its major versions: see `sdl` and `sdl2`, `libpng*` etc.)

Case in point: the current official electron package is `1.4.16-2`, whereas itch v23.4.1 uses electron `1.6.2`, and the current itch v25 branch uses `1.7.3` (a beta release).

There's more than just javascript APIs - you could expect only parts of the app to be broken by a mismatched electron version. But now, itch ships with binary modules, such as `sqlite3` and [node-shoom](https://github.com/itchio/node-shoom). Those will simply not load if the ABI versions aren't exact matches.

For all those reasons, I won't be devoting time & resources to maintain an official version of itch that doesn't ship its own electron runtime. What we *might* do is eventually build our own electron binaries, potentially stripping them from stuff we don't need. But there's not a lot we don't need, since the app is also a fully-featured web browser.

Hopefully that makes sense!

**Comment by quyse** [comment]
Thanks for your thoughtful consideration! I can say that the initial thought was just to save some space (and maybe make it a bit faster, because shared electron binary may be already loaded in memory), but then I thought that maybe it can also help bring itch app to Arch/Manjaro as a package in official repositories. I wouldn't mind if it lags behind the latest version by a few days.

Honestly I like how Linux distributions work, with maintainers controlling versions and checking compatibility between shared packages, as opposed to self-updated self-contained portable apps, and I think a lot of users like that as well. (And I know you don't like package managers). On Arch packages are usually (almost) up-to-date with upstream, but it seems there're problems with particular `electron` package though, as it wasn't updated for a while (and marked out-of-date), probably because multiple packages depend on it, and update of which must be coordinated.

So yeah, considering how quickly itch app is moving, and lack of backward compatibility in electron, seems like this PR's idea is a bad idea. Maybe it can be considered again if electron's situation changes, i.e. API becomes stable enough :)

**Comment by fasterthanlime** [comment]
> Maybe it can be considered again if electron's situation changes, i.e. API becomes stable enough :)

There is hope, sorta, because of this: https://github.com/nodejs/abi-stable-node

Right now though, I sleep much better at night knowing that the plan is self-sufficient self-update system that's deployed the same way to all 3 OSes, without having to go through any of the (multitudes) of packaging conventions and processes.
